docs/developers/extending.rst

```diff
@@ -0,0 +1,157 @@
+
+Extensions cannot require modifications to the Django settings of Open Forms. Any run-time configuration option can
+be specified as:
+
+#. Deployment time environment variables. See `plugin.py:35 <https://github.com/open-formulieren/demo-extension/blob/main/demo_extension/plugin.py#L35>`_.
+
+#. Dynamic options using ``django-solo``, with the advantage that configuration can be
+   modified at runtime through the Admin interface. See `models.py`_ and `admin.py`_ in the demo plugin.
+   If you add a solo model, you will have to generate migrations.
+   See the :ref:`section below<test_extension_plugin_locally>` for details.
+
+#. For certain types of plugins, like for registration backends, it is also possible to add form-specific options.
+   These can be specified with a serializer. See `config.py`_ for details.
+
+If any initialization or module loading is needed, this should be done as part of the
+extension's ``extension.apps.ExtensionConfig.ready`` hook, which is called when Django
+bootstraps. This is the mechanism by which Open Forms plugins are registered
+(refer to the :ref:`plugin section<plugins_index>` for further details).
+
+.. _models.py: https://github.com/open-formulieren/demo-extension/blob/main/demo_extension/models.py
+.. _admin.py: https://github.com/open-formulieren/demo-extension/blob/main/demo_extension/admin.py
+.. _config.py: https://github.com/open-formulieren/demo-extension/blob/main/demo_extension/config.py
+
+.. _test_extension_plugin_locally:
+
+Testing locally
+---------------
+
+In order to test your external plugin locally, you can use symlinks. Inside the ``src/`` directory of Open Forms,
+create a symlink to your package. For example, in the case of the demo extension package:
+
+.. code-block:: bash
+
+    ln -s /path/to/plugin/demo_extension .
+
+Now add an environment variable ``OPEN_FORMS_EXTENSIONS`` with the name of your package. The variable is a
+comma-separated list of valid python identifiers (i.e. the python package names). For example:
+
+.. code-block:: bash
+
+    export OPEN_FORMS_EXTENSIONS=demo_extension
+
+The Open Forms ``base.py`` settings module will extract this configuration value and
+append the list of packages to ``INSTALLED_APPS``, which causes them to be loaded by
+Django.
+
+If you need to generate migrations for your package, you can now do it as follows (from within the Open Forms directory):
+
+.. code-block:: bash
+
+    python src/manage.py makemigrations demo_extension
+    python src/manage.py migrate
+
+If you created a solo model, you can add the configuration page to the Admin. To do this, login into the Open Forms
+Admin:
+
+#. Go to the **Configuratie** > **Application groups**.
+
+#. Click on **Configuratie**.
+
+#. In the **Models** section, look for the name of your configuration model in the left table (for the demo extension, this was  ``demo_extension.Demoextensionconfig``).
+   Then double click on it to add it to the right table.
+
+#. Save the configuration.
```

Comment:
We'll have to find a better solution for this, as I think the post-migrate signal handlers deletes any manual changes and restores the fixture configuration.